src/core/simulation_sequence.jl

```diff
@@ -51,77 +47,46 @@ function _fill_execution_order(
     execution_order::Vector{Int},
     interval_run_counts::Dict{Int, Int},
 )
-    function _fill_stage(index::Int, stage::Int)
-        if stage < last_stage
-            next_stage = stage + 1
-            for i in 1:interval_run_counts[next_stage]
-                index = _fill_stage(index, next_stage)
+    function _fill_problem(index::Int, problem::Int)
+        if problem < last_problem
+            next_problem = problem + 1
+            for i in 1:interval_run_counts[next_problem]
+                index = _fill_problem(index, next_problem)
             end
         end
-        execution_order[index] = stage
+        execution_order[index] = problem
         index -= 1
```

Comment:
What is the point of `index-=1` here?

Reply:
This calculates the sequence from the end through the beggining